libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork.m

```diff
@@ -93,7 +93,7 @@ + (instancetype)currentNetwork {
 
 + (instancetype)networkForUserAccount:(SFUserAccount*)account {
     CSFNetwork *instance = nil;
-    if (!account.isTemporaryUser) {
+    if (account!=nil) {
```

Comment:
In that case, would cachedNetworkForUserAccount just return nil for the temporary user ending with the same outcome?